src/attack.c

```diff
@@ -731,13 +731,30 @@ attack_start (struct attack_start_args *args)
 }
 
 static void
-apply_hosts_preferences (gvm_hosts_t *hosts)
+apply_hosts_excluded (gvm_hosts_t *hosts)
 {
-  const char *ordering = prefs_get ("hosts_ordering"),
-             *exclude_hosts = prefs_get ("exclude_hosts");
+  const char *exclude_hosts = prefs_get ("exclude_hosts");
+
+  /* Exclude hosts ? */
+  if (exclude_hosts)
+    {
+      /* Exclude hosts, resolving hostnames. */
+      int ret = gvm_hosts_exclude (hosts, exclude_hosts);
+
+      if (ret > 0)
+        g_message ("exclude_hosts: Skipped %d host(s).", ret);
+      if (ret < 0)
+        g_message ("exclude_hosts: Error.");
+    }
 
   if (hosts == NULL)
```

Comment:
Does not seem useful here. We return nonetheless. Should probably be at the top of the function.
This check might be needed at the top of the other new functions as well?